cli/command/completion/functions.go

```diff
@@ -189,3 +194,122 @@ var commonPlatforms = []string{
 func Platforms(_ *cobra.Command, _ []string, _ string) (platforms []string, _ cobra.ShellCompDirective) {
 	return commonPlatforms, cobra.ShellCompDirectiveNoFileComp
 }
+
+type ImageSearchResult struct {
+	ID        string    `json:"id"`
+	Name      string    `json:"name"`
+	Slug      string    `json:"slug"`
+	Type      string    `json:"type"`
+	CreatedAt time.Time `json:"created_at"`
+	UpdatedAt time.Time `json:"updated_at"`
+	ShortDesc string    `json:"short_description"`
+	Source    string    `json:"source"`
+	StarCount int       `json:"star_count"`
+}
+
+type ImageSearch struct {
+	Totals  int                 `json:"totals"`
+	Results []ImageSearchResult `json:"results"`
+}
+
+type Image struct {
+	ID          int       `json:"id"`
+	Name        string    `json:"name"`
+	TagStatus   string    `json:"tag_status"`
+	V2          bool      `json:"v2"`
+	Digest      string    `json:"digest"`
+	LastUpdated time.Time `json:"last_updated"`
+	LastUpdater int       `json:"last_updater"`
+	Creator     int       `json:"creator"`
+	Repository  int       `json:"repository"`
+}
+
+type ImageTags struct {
+	Count   int     `json:"count"`
+	Next    string  `json:"next"`
+	Prev    string  `json:"prev"`
+	Results []Image `json:"results"`
+}
+
+func RemoteImages(cmd *cobra.Command, arg []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	ctx := cmd.Context()
+	c := &http.Client{
+		Timeout: 2 * time.Second,
+	}
+
+	if imageName, imageTag, ok := strings.Cut(toComplete, ":"); ok {
+		u, err := url.Parse("https://hub.docker.com/v2/repositories/library/" + imageName + "/tags/")
+		if err != nil {
+			logrus.Errorf("Error parsing hub image tags URL: %v", err)
+			return nil, cobra.ShellCompDirectiveError
+		}
+		q := u.Query()
+		q.Set("ordering", "last_updated")
+		q.Set("page_size", "25")
+		q.Set("name", imageTag)
+		u.RawQuery = q.Encode()
+
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+		if err != nil {
+			logrus.Errorf("Error creating hub image tags request: %v", err)
```

Comment:
I honestly don't know what the best way would be for us to track errors for the completions. Maybe we should write debug logs to a file?